### PR TITLE
Fixed occasional KeyNotFoundExceptions coming from DefaultPointerMedi…

### PR DESCRIPTION
**Pull request by marek-stoj**
…ator

## Changes
- Fixes: # .

### CODE REVIEW

File: Assets/MRTK/SDK/Features/UX/Scripts/Pointers/DefaultPointerMediator.cs [hunk 1]
```diff
@@ -142,7 +142,7 @@ public virtual void UpdatePointers()
                         pointer.IsActive = true;
                         unassignedPointers.Remove(pointer);
 
-                        if (pointer.InputSourceParent != null)
+                        if (pointer.InputSourceParent != null && pointerByInputSourceParent.ContainsKey(pointer.InputSourceParent))
```
Comment (keveleigh) [inline Assets/MRTK/SDK/Features/UX/Scripts/Pointers/DefaultPointerMediator.cs:145]: Another optimization could be using `TryGetValue` here, so we don't continually index in the `foreach`

Comment (keveleigh) [inline Assets/MRTK/SDK/Features/UX/Scripts/Pointers/DefaultPointerMediator.cs:145]: Another optimization could be using `TryGetValue` here, so we don't continually index in the `foreach`

**Comment by RogPodge** [comment]
small ping on this @marek-stoj

File: Assets/MRTK/SDK/Features/UX/Scripts/Pointers/DefaultPointerMediator.cs (hunk 1 above)
Comment (marek-stoj) [reply Assets/MRTK/SDK/Features/UX/Scripts/Pointers/DefaultPointerMediator.cs]: Done

Comment (marek-stoj) [reply Assets/MRTK/SDK/Features/UX/Scripts/Pointers/DefaultPointerMediator.cs]: Done

**Review by RogPodge — APPROVED** [review]

**Pull request merged**